src/cr/cube/slices.py

```diff
@@ -0,0 +1,1140 @@
+        return self._counts
+
+    @lazyproperty
+    def base_values(self):
+        if not isinstance(self._base_counts, np.ndarray):
+            return np.array([self._base_counts])
+        return self._base_counts
+
+    @lazyproperty
+    def base(self):
+        return np.sum(self._base_counts)
+
+    @lazyproperty
+    def margin(self):
+        return np.sum(self._counts)
+
+    @lazyproperty
+    def table_margin(self):
+        return self._table_margin
+
+    @lazyproperty
+    def proportions(self):
+        return self.values / self.margin
+        # return self.values / self.base
+
+    @lazyproperty
+    def table_proportions(self):
+        return self.values / self.table_margin
+
+    @lazyproperty
+    def pruned(self):
+        # return self.margin == 0
+        return self.base == 0 or np.isnan(self.base)
+
+
+class _MultipleResponseVector(_CategoricalVector):
+    @lazyproperty
+    def values(self):
+        return self._selected
+
+    @lazyproperty
+    def base_values(self):
+        return self._base_counts[0, :]
+
+    @lazyproperty
+    def base(self):
+        counts = zip(self._selected_unweighted, self._not_selected_unweighted)
+        return np.array(
+            [selected + not_selected for (selected, not_selected) in counts]
+        )
+
+    @lazyproperty
+    def margin(self):
+        counts = zip(self._selected, self._not_selected)
+        return np.array(
+            [selected + not_selected for (selected, not_selected) in counts]
+        )
+
+    @lazyproperty
+    def pruned(self):
+        # return np.all(self.margin == 0)
+        return np.all(self.base == 0) or np.all(np.isnan(self.base))
+
+    @lazyproperty
+    def _selected(self):
+        return self._counts[0, :]
+
+    @lazyproperty
+    def _not_selected(self):
+        return self._counts[1, :]
+
+    @lazyproperty
+    def _selected_unweighted(self):
+        return self._base_counts[0, :]
+
+    @lazyproperty
+    def _not_selected_unweighted(self):
+        return self._base_counts[1, :]
+
+
+class Insertions(object):
+    def __init__(self, dimensions, slice_):
+        self._dimensions = dimensions
+        self._slice = slice_
+
+    @lazyproperty
+    def _row_dimension(self):
+        return self._dimensions[0]
+
+    @lazyproperty
+    def _column_dimension(self):
+        return self._dimensions[1]
+
+    @lazyproperty
+    def row_anchors(self):
+        return tuple(row.anchor for row in self.rows)
+
+    @lazyproperty
+    def column_anchors(self):
+        return tuple(column.anchor for column in self.columns)
+
+    @lazyproperty
+    def _rows(self):
+        if self._row_dimension.dimension_type in (DT.MR, DT.CA):
+            return tuple()
+
+        return tuple(
+            InsertionRow(self._slice, subtotal)
+            for subtotal in self._row_dimension._subtotals
+        )
+
+    @lazyproperty
+    def _columns(self):
+        if len(self._dimensions) < 2 or self._column_dimension.dimension_type in (
```

Comment:
This can probably be taken care of with inheritance, but it's relatively simple (still) so I just kept it...